pkg/errors/conversion.go

```diff
@@ -49,6 +49,7 @@ func From(err error) (out *Error, ok bool) {
 		return &e, true
 	case ErrorDetails: // Received over an API.
 		var e Error
+		e.code = err.Code()
```

Comment:
This is in the wrong place. Setting things from `ErrorDetails` into `Error` is done in `setErrorDetails`.